api-reference/10 UI Components/dxDataGrid/1 Configuration/export/formats.md

```diff
@@ -6,7 +6,368 @@ default: 'DataGrid'
+                doc.save('Customers.pdf');
+            })
+        }
+    }
+
+    <!-- tab: app.module.ts -->
+    import { BrowserModule } from '@angular/platform-browser';
+    import { NgModule } from '@angular/core';
+    import { AppComponent } from './app.component';
+
+    import { DxDataGridModule } from 'devextreme-angular';
+
+    @NgModule({
+        declarations: [
+            AppComponent
+        ],
+        imports: [
+            BrowserModule,
+            DxDataGridModule
+        ],
+        providers: [ ],
+        bootstrap: [AppComponent]
+    })
+    export class AppModule { }
+
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <div>
+            <DxDataGrid ...
+                @exporting="onExporting"
+            >
+                <!-- ... -->
+                <DxExport
+                    :enabled="true"
+                    :formats="['pdf']"
+                />
+            </DxDataGrid>
+        </div>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.light.css';
+    import { DxDataGrid, DxExport } from 'devextreme-vue/data-grid';
+    import { jsPDF } from 'jspdf';
+    import { exportDataGrid } from 'devextreme/pdf_exporter';
+
+    export default {
+        components: {
+            DxDataGrid
+        }
+        data() {
+            return {
+                // ...
+            };
+        },
+        methods: {
+            onExporting(e) {
+                const doc = new jsPDF();
+                exportDataGrid({
+                    jsPDFDocument: doc,
+                    component: e.component,
+                }).then(() => {
+                    doc.save('Companies.pdf');
+                });
+            },
+        },
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import 'devextreme/dist/css/dx.light.css';
+    import DataGrid, { Export } from 'devextreme-react/data-grid';
+    import { jsPDF } from 'jspdf';
+    import { exportDataGrid } from 'devextreme/pdf_exporter';
+
+    const exportFormats = ['pdf'];
+
+    export default function App() {
+        const onExporting = React.useCallback((e) => {
+            const doc = new jsPDF();
+            exportDataGrid({
+                jsPDFDocument: doc,
+                component: e.component,
+            }).then(() => {
+                doc.save('Companies.pdf');
+            });
+        });
+        return (
+            <React.Fragment>
+                <div>
+                    <DataGrid ...
+                        onExporting={onExporting}
+                    >
+                        {/* ... */}
+                        <Export enabled={true} formats={exportFormats}>
+                    </DataGrid>
+                </div>
+            </React.Fragment>
+        );
+    }
+
+---
+
+You can specify multiple formats: `['xlsx', 'pdf', string]`. The `string` format specifies the custom text shown in the **Export** button.
```

Comment:
I think it's better to replace `string` with `'someOtherFormat'` or something like this. `string` is just typescript type and for me it's confusing to see it as value

Reply:
I agree. And I also wanted to note that there are two things going on at once, and I would rather not mix them up. One thing - you can specify a custom format label, such as "CSV". Second thing - you can specify multiple labels. So I suggest we explain these things on step at a time: 

```suggestion
The predefined values for the **formats** property are 'xlsx' and 'pdf'. You can also specify any custom string, such as `['CSV']`. If you do that, the export command caption will read "Export all data (selected rows) to CSV". Once again, you'll need to change the onExporting handler to produce a file in that format. 

Since the **formats** property accepts an array, you can specify multiple formats. For example, you can set the property to  `['pdf', 'CSV']`. In this case, the grid displays multiple export commands: "Export all data (selected rows) to PDF" and "Export all data (selected rows) to CSV". In the onExporting handler, add logic that checks which button initiated the export operation.
```

I assume that you need to specify CSV in capital letters... Or does the grid automatically capitalize custom strings? 

You may also want to mention the properties that specify the templates for command captions.

Reply:
It's better to write `'csv'`. Export button title will be "Export all data to CSV" , `onExporting(e)` event's `e.format` will be `'csv'`